src/parser.js

```diff
@@ -0,0 +1,89 @@
+
```

Comment:
Finally, it makes sense to have a configuration object that uses a parser (and not the contrary).
All other classes should interact with the configuration obj and not the parser directly